.circleci/config.yml

```diff
@@ -1,92 +1,118 @@
+            export DOCKER_BUILDKIT=1
+            export DOCKER_CLI_EXPERIMENTAL=enabled
+            echo 'export BUILDKIT_PROGRESS=plain' >> $BASH_ENV
+            echo 'export DOCKER_BUILDKIT=1' >> $BASH_ENV
+            echo 'export DOCKER_CLI_EXPERIMENTAL=enabled' >> $BASH_ENV
+
+            curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -
+            sudo add-apt-repository \
+              "deb [arch=amd64] https://download.docker.com/linux/ubuntu \
+              $(lsb_release -cs) \
+              stable"
+
+            sudo apt-get update
+            sudo apt-get install -y \
+              containerd.io \
+              docker-ce \
+              docker-ce-cli
+            sudo service docker restart
+
+  deploy_release_packages:
+    description:  >
+      This will build and publish release packages for tag "$CIRCLE_TAG"
+    steps:
+      - run:
+          name: Deploy Release Packages
+          command:  |
+            ./build.sh --debug --clean --generate --package --package-udfs --upload --bucket=dl.influxdata.com/kapacitor/releases --platform=all --arch=all --release
 
-      run_tests:
-            description:  >
-                  This will run Kapacitor Tests
-            steps:
-                  - run:
-                        name: Running Kapacitor Tests
-                        no_output_timeout: 30m
-                        command:  |
-                              ./circle-test.sh
-      deploy_nightly:
-            description:  >
-                    This will build and publish nightly releases
-            steps:
-                  - run:
-                        name: Deploy Nightly Build
-                        command:  |
-                                ./build.sh --debug --clean --generate --package --package-udfs --upload --bucket=dl.influxdata.com/kapacitor/releases/nightly --platform=all --arch=all --nightly
+  run_tests:
+    description:  >
+      This will run Kapacitor Tests
+    steps:
+      - run:
+          name: Running Kapacitor Tests
+          no_output_timeout: 30m
+          command:  |
+            ./circle-test.sh
+  deploy_nightly:
+    description:  >
+      This will build and publish nightly releases
+    steps:
+      - run:
+          name: Deploy Nightly Build
+          command:  |
+            ./build.sh --debug --clean --generate --package --package-udfs --upload --bucket=dl.influxdata.com/kapacitor/releases/nightly --platform=all --arch=all --nightly
 
 jobs:
-      build:
-            docker:
-                  - image: nathanielc/docker-client
-            parallelism: 2
-            working_directory: ~/kapacitor
-            steps:
-                  - checkout
-                  - setup_remote_docker:
-                        docker_layer_caching: true
-                  - run_tests
+  build:
+    executor: linux-amd64
+    parallelism: 2
+    working_directory: ~/kapacitor
+    steps:
+      - checkout
+      - upgrade_docker
```

Comment:
I think you should be able to request a specific VM image and avoid the docker upgrade

Reply:
:+1: the image I've pinned in the `executors` definition is the same as used in influxdb 2.x / influx-cli, and I remember its pre-installed Docker being pretty old. I'll check if there's a newer image out that we can try.